sympy/matrices/matrices.py

```diff
@@ -78,2091 +83,224 @@ def __repr__(self):
 
 
 class MatrixDeterminant(MatrixCommon):
-    """Provides basic matrix determinant operations.
-    Should not be instantiated directly."""
+    """Provides basic matrix determinant operations. Should not be instantiated
+    directly. See ``determinant.py`` for their implementations."""
+
+    _find_reasonable_pivot       = _find_reasonable_pivot
+    _find_reasonable_pivot_naive = _find_reasonable_pivot_naive
+    _eval_determinant            = _det
+    adjugate                     = _adjugate
+    charpoly                     = _charpoly
+    cofactor                     = _cofactor
+    cofactor_matrix              = _cofactor_matrix
+    det                          = _det
+    det_bareiss                  = _det_bareiss
```

Comment:
I would prefer explicit methods here like
https://github.com/sympy/sympy/blob/3598399232132234bc646acc6ecb690a093d9d63/sympy/core/expr.py#L3583-L3596
Although ideally the docstring should be written so that it generates a link to the referenced function in the html docs.

Reply:
This is something I specifically asked about in #18253 but since nobody commented I went ahead with the assignment. Can change it back easily though if you want.